agent/consul/state/config_entry.go

```diff
@@ -344,6 +344,27 @@ func deleteConfigEntryTxn(tx WriteTxn, idx uint64, kind, name string, entMeta *a
 			return fmt.Errorf("failed updating gateway-services index: %v", err)
 		}
 	}
+
+	c := existing.(structs.ConfigEntry)
+	switch c.GetKind() {
+	case structs.ServiceDefaults:
+		if c.(*structs.ServiceConfigEntry).Destination != nil {
+			gsKind, err := GatewayServiceKind(tx, sn.Name, &sn.EnterpriseMeta)
+			if err != nil {
+				return fmt.Errorf("failed updating gateway mapping: %s", err)
+			}
+			if gsKind == structs.GatewayservicekindDestination {
+				gsKind = structs.GatewayservicekindUnknown
+			}
```

Comment:
Feel free to ignore this one but I didn't understand this check. Maybe there is a case where `service-default` has a `destination` but is mislabeled? Then would it be:
```suggestion
			if gsKind == structs.GatewayservicekindUnknown {
				gsKind = structs.GatewayservicekindDestination
			}
```

Reply:
So this is because the config entry is not yet deleted, so when we look it up we find it as a Destination type (potentially, if a service with the same name do not exist) so in that case we set it to unknown kind because we are deleting the config entry just after.